File: 2023/day1/day1.py
file = open("2023/day1/data.txt", "r")
input = file.readlines()

sum = 0
for line in input:
    curr = ""
    last = ""
    first = True
    for char in line:
        if char.isdigit() and first:
            curr += str(char)
            last = str(char)
            first = False
        elif char.isdigit():
            last = str(char)
    curr += str(last)
    sum += int(curr)

print(sum)

File: 2025/day1/day1.java
package day1;

import java.io.*;
import java.util.*;

public class day1 {
    static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new FileReader("2025/day1/data.txt"));
        String line;
        List<String> input = new ArrayList<>();

        int max = 100;
        int curr = 50;
        int count = 0;

        while ((line = br.readLine()) != null) {
            input.add(line);
        }

        for (String s : input) {
            char dir = s.charAt(0);
            int size = Integer.parseInt(s.substring(1));

            if (dir == 'L') curr = modulus((curr - size), max);
            else curr = modulus((curr + size), max);
            
            if (curr == 0) count++;
        }

        System.out.println(count);
        br.close();
    }

    private static int modulus(int l, int r) {
        return (((l % r) + r) % r);
    }
}